ray-operator/controllers/ray/rayservice_controller.go

```diff
@@ -1028,8 +1030,13 @@ func (r *RayServiceReconciler) updateStatusForActiveCluster(ctx context.Context,
 		return err
 	}
 
+	headSvcName, err := utils.GenerateHeadServiceName(utils.RayServiceCRD, rayClusterInstance.Spec, rayClusterInstance.Name)
```

Comment:
We should have a more elegant method to initialize the dashboard client. Calling GenerateHeadServiceName every time we want to initialize the client makes the codebase complex and hard to maintain.

Reply:
Agreed, I thought this was kind of ugly too. Maybe we can pass the entire RayCluster to InitClient?

Reply:
> Maybe we can pass the entire RayCluster to InitClient?

I can make this change in this PR if you want

Reply:
> I can make this change in this PR if you want

Thanks! It is helpful! Maybe we can consider adding the head service's name to RayCluster's status in the future.

Reply:
I updated InitClient to receive a `RayCluster` so we can call GenerateHeadServiceName from inside `InitClient`. This requires InitClient to now return an error